clicommand/pipeline_upload.go

```diff
@@ -280,19 +281,12 @@ var PipelineUploadCommand = cli.Command{
 			}
 		}
 
-		// By default, when in ApplyIfChanged mode, `if_changed` attributes are
-		// stripped.
-		var stepsMutation func(pipeline.Steps)
-		if cfg.ApplyIfChanged {
-			// If we can figure out which files have changed, we can apply
-			// `if_changed` filtering.
-			changedPaths, err := gatherChangedFiles(cfg.GitDiffBase)
-			if err != nil {
-				l.Error("Couldn't determine git diff from upstream, not skipping any pipeline steps: %v", err)
-				stepsMutation = stripIfChanged
-			} else {
-				stepsMutation = func(steps pipeline.Steps) { applyIfChanged(l, steps, changedPaths) }
-			}
+		ifChanged := &ifChangedApplicator{
+			enabled: cfg.ApplyIfChanged,
+			// An empty string for the flag can happen if the default env var
+			// ends up with an empty string (e.g. this build is not a PR build
+			// ...yet.)
+			diffBase: cmp.Or(cfg.GitDiffBase, "origin/main"),
```

Comment:
https://pkg.go.dev/cmp#Or somehow I hadn't seen `cmp.Or` before! 😮 Gosh, there is a bunch of noisey code I could have cleaned up with that! 😅